argo/src/Argo/DefaultMain.hs

```diff
@@ -166,18 +173,20 @@ realMain theApp opts =
   case transportOpt opts of
     StdIONetstring ->
       serveStdIONS theApp
-    SocketNetstring sessionOpt publicityOpt portOpt ->
+    SocketNetstring sessionOpt publicityOpt portOpt hostNameOpt ->
       do sessionResult <- getOrLockSession sessionOpt publicityOpt portOpt
+         let hostname = case hostNameOpt of
```

Comment:
As a matter of style, I very weakly prefer `fromMaybe` for this pattern. But it's not a big deal.